File: dodam-application/dodam-rest-api/src/main/java/b1nd/dodam/restapi/schedule/presentation/ScheduleController.java
package b1nd.dodam.restapi.schedule.presentation;

import b1nd.dodam.restapi.schedule.application.ScheduleUseCase;
import b1nd.dodam.restapi.schedule.application.data.req.ScheduleReq;
import b1nd.dodam.restapi.schedule.application.data.res.ScheduleRes;
import b1nd.dodam.restapi.support.data.Response;
import b1nd.dodam.restapi.support.data.ResponseData;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.time.LocalDate;
import java.util.List;

@RestController
@RequestMapping(value = "/schedule")
@RequiredArgsConstructor
public class ScheduleController {

    private final ScheduleUseCase useCase;

    @PostMapping
    public Response create(@RequestBody @Valid ScheduleReq req) {
        return useCase.create(req);
    }

    @PatchMapping("/{id}")
    public Response modify(@PathVariable int id, ScheduleReq req) {
        return useCase.modify(id, req);
    }

    @DeleteMapping("/{id}")
    public Response delete(@PathVariable int id) {
        return useCase.delete(id);
    }

    @GetMapping("/today")
    public ResponseData<List<ScheduleRes>> getToday() {
        return useCase.getToday();
    }

    @GetMapping("/date")
    public ResponseData<List<ScheduleRes>> getByDate(@RequestParam int year,
                                                     @RequestParam int month,
                                                     @RequestParam int day) {
        return useCase.getByDate(year, month, day);
    }

    @GetMapping("/search")
    public ResponseData<List<ScheduleRes>> getByDate(@RequestParam LocalDate startAt,
                                                     @RequestParam LocalDate endAt) {
        return useCase.getByDate(startAt, endAt);
    }

    @GetMapping("/search/keyword")
    public ResponseData<List<ScheduleRes>> searchByName(@RequestParam("keyword") String keyword) {
        return useCase.searchByName(keyword);
    }

    @GetMapping
    public ResponseData<List<ScheduleRes>> getAllOrderByIdDesc(@RequestParam int page,
                                                  @RequestParam int size) {
        return useCase.getAllOrderByIdDesc(page, size);
    }

    @PostMapping("/neis")
    public Response createByNeis(
            @RequestParam LocalDate startDate,
            @RequestParam LocalDate endDate
    ){
        return useCase.createScheduleByNeis(startDate, endDate);
    }

}

File: services/service-neis/src/main/kotlin/com/b1nd/dodamdodam/neis/presentation/schedule/ScheduleController.kt
package com.b1nd.dodamdodam.neis.presentation.schedule

import com.b1nd.dodamdodam.core.security.annotation.authentication.UserAccess
import com.b1nd.dodamdodam.core.security.passport.enumerations.RoleType
import com.b1nd.dodamdodam.neis.application.schedule.ScheduleUseCase
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import java.time.LocalDate

@RestController
@RequestMapping("/schedule")
class ScheduleController(
    private val scheduleUseCase: ScheduleUseCase,
) {
    @UserAccess
    @GetMapping("/me")
    fun getMySchedules() = scheduleUseCase.getMyWeeklySchedules()

    @UserAccess
    @GetMapping
    fun getSchedules(@RequestParam date: LocalDate, @RequestParam grade: Int, @RequestParam room: Int) =
        scheduleUseCase.getSchedulesByDate(date, grade, room)

    @UserAccess
    @GetMapping("/all")
    fun getAllSchedules(@RequestParam date: LocalDate) =
        scheduleUseCase.getAllSchedulesByDate(date)

    @UserAccess(roles = [RoleType.ADMIN])
    @PostMapping("/sync/weekly")
    fun syncWeeklySchedules(@RequestParam mondayDate: LocalDate) =
        scheduleUseCase.syncWeeklySchedules(mondayDate)

    @UserAccess(roles = [RoleType.ADMIN])
    @PostMapping("/sync/daily")
    fun syncDailySchedules(@RequestParam date: LocalDate) =
        scheduleUseCase.syncDailySchedules(date)
}